Make wiselink work again

geoffreykwan opened the pull request:
Test that wiselink works.

Scenarios to test
- link to another step
- link to a component in another step
- link to a component in the current step
- button to another step
- button to a component in another step
- button to a component in the current step

Closes #80

geoffreykwan committed:
Make wiselink work again. #80

hirokiterashima committed:
Merge branch 'develop' into issue-80-make-wiselink-work-again

hirokiterashima commented on src/assets/wise5/components/html/html-student/html-student.component.ts:
```diff
@@ -34,11 +37,84 @@ export class HtmlStudent extends ComponentStudent {
+  }
+
+  goToNode(nodeId: string, componentId: string): void {
+    this.currentNodeChangedSubscription = this.StudentDataService.currentNodeChanged$.subscribe(
+      () => {
+        if (componentId != '') {
+          this.scrollToComponentAndHighlight(componentId);
+        }
+      }
+    );
+    this.StudentDataService.endCurrentNodeAndSetCurrentNodeByNodeId(nodeId);
```
Should the subscription and scroll/highlight occur outside of html-student component, like in a service? When the currentNode changes, this html-student component should no longer be alive, right? It seems strange that this subscription is alive and participating in events even after the user has navigated away from the component.

hirokiterashima reviewed (changes requested):
It looks like the highlight does not go away if the link is to the component that is on the same step.

Also, I wrote a question about code inline. Let me know what you think.

geoffreykwan committed:
Make wiselink only fire within the scope of that component. Move the currentNodeChanged subscription to nodeService.ts. Fix a problem with wiselink elements not working after you modify the html. #80

geoffreykwan commented:
The listener was being fired once for each HTML component on the step. Now the wiselinkclicked only fires once within that component. This fixed the problem of highlighting not going away.

We now listen for the currentNodeChanged event in the nodeService.ts.

I also fixed a bug where if you created a wiselink and then modified the html, the wiselink attribute would be lost and the wiselink would no longer work.

hirokiterashima committed:
Removed unused css('transition') call.

It doesn't seem to be doing anything new. Things work without it. #80

hirokiterashima committed:
Added back the css('transition').

It is indeed necessary when linking to the same step as the link/button.
Highlighting the second time won't work without it.
Updated comments to explain this special case.

hirokiterashima reviewed (approved):
New changes look good.
